### PR TITLE
Tweak path related tests

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Needed to update a few tests for unix whitespace behavior.

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
These are hitting in https://github.com/dotnet/corefx/pull/22368

File: src/CoreFx.Private.TestUtilities/tests/AssertExtensionTests.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Sdk;
+
+namespace CoreFx.Private.TestUtilities.Tests
+{
+    public class AssertExtensionTests
```
Comment (ianhays) [inline src/CoreFx.Private.TestUtilities/tests/AssertExtensionTests.cs]: Are these unit tests for the functionality of our test helpers? That shouldn't be necessary, and if it is then our helpers are too complex.

Comment (JeremyKuhne) [inline src/CoreFx.Private.TestUtilities/tests/AssertExtensionTests.cs]: Test helpers should always be tested- it is part of our infrastructure. What you don't need to do is test test methods.

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs [hunk 2]
```diff
@@ -35,13 +35,28 @@ public static void ChangeExtension(string path, string newExtension, string expe
...
-        public static void GetDirectoryName_EmptyOrWhitespace_Throws(string path)
+        [Fact]
+        public static void GetDirectoryName_EmptyThrows()
+        {
+            AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(string.Empty));
+        }
+
+        [Theory,
+            InlineData(" "),
+            InlineData("\r\n")]
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: I prefer having separate attributes, but it seems we aren't consisetnt...

File: src/CoreFx.Private.TestUtilities/tests/AssertExtensionTests.cs (hunk 1 above)
Comment (JeremyKuhne) [inline src/CoreFx.Private.TestUtilities/tests/AssertExtensionTests.cs]: To be completely clear, where plausible and particularly in shared helpers.

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs [hunk 3]
```diff
@@ -35,13 +35,28 @@ public static void ChangeExtension(string path, string newExtension, string expe
...
+        [ActiveIssue(21358)] // Pending CoreCLR behavior change
+        public static void GetDirectoryName_SpaceOrControlCharsThrowOnWindows(string path)
         {
-            AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(path));
+            Action action = () => Path.GetDirectoryName(path);
+            if (PlatformDetection.IsWindows)
```
Comment (ianhays) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: This should be using PlatformSpecific attributes instead.

Also the test name has "Windows" in it but runs on Unix.

Comment (JeremyKuhne) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: I don't agree in this case. This particular case is saying "throws on Windows but not Unix". Same input data. I could make the test name longer, but I think the fact that it isn't Windows only makes it implicit that it is "WindowsAndNotUnix".

I could split this into two entry points and shared test data property, but that seems like over-kill to me in this case. Same goes for duplicating the method.  I think it is more coherent to have both sides of the coin together here.

File: src/CoreFx.Private.TestUtilities/tests/AssertExtensionTests.cs (hunk 1 above)
Comment (JeremyKuhne) [inline src/CoreFx.Private.TestUtilities/tests/AssertExtensionTests.cs]: The regression risk should also be considered. In these extensions it is plausible for an error in logic to not be found immediately, potentially masking test failures for extended periods of time. We have a history of our test infrastructure doing just that.

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs (hunk 2 above)
Comment (ianhays) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: ditto

**Comment by steji113** [comment]
Refer to #8655 for background on changes.

**Comment by karelz** [comment]
@JeremyKuhne FYI: Fixed the milestone to 2.1. The change didn't make it into 2.0 branch.
If you think it is needed in 2.1 servicing branch, we will need to port it - please file a new tracking issue in such case, thanks!

**Pull request merged**